### ISSUE TITLE
[Bug]: ANCOVA return different results compared to SAS

### ISSUE THREAD

**Issue by JiaLiu0001** [insightsengineering/tern#681]
### What happened?

ANCOVA model with treatment, visit, and treatment * visit interaction effect as fixed effects, and baseline as covariates. Different source data and tools might cause different results.

Methods in different tools:
SAS: Proc mixed
R: Ancova tern package ([AVOT01](https://docs.roche.com/doc/NEST%20Exploratory%20Releases/2022_01_28/tlg-catalog/)) 

Take treatment vs. CONTROL, Week 24 result as an example. The following are 3 scenarios to fit the model:

1. Two treatments: TRT & CONTROL. One visit: Week 24  (R and SAS have exactly the same results)
(This scenario is dropped out as only 2 treatments are included. According to Ruchi, under this situation, ANCOVA is not the first choice and using linear regression should make more sense)
SAS Result: (fixed effect: treatment, visit, and treatment*visit)
Adjust means: -1.4906
Difference in adjust means: -1.4086. 95%CI (-1.6070,-1.2103)
R Result:
Adjust means: -1.49
Difference in adjust means: -1.41. 95%CI (-1.61,-1.21)

2. Multiple treatments: TRT2/3/4/6 & CONTROL. One visit: Week 24 (R and SAS have exactly the same results)
SAS Result: (fixed effect: treatment, visit, and treatment*visit)
Adjust means: -1.4855
Difference in adjust means: -1.4071. 95%CI (-1.7373,-1.0768)
SAS model that has only treatment as the fixed effect
Adjust means: -1.4855
Difference in adjust means: -1.4071. 95%CI (-1.7373,-1.0768)
R Result:
Adjust means: -1.49
Difference in adjust means: -1.41. 95%CI (-1.74,-1.08)

# 3. Multiple treatments: TRT2/3/4/6 & CONTROL. Multiple visits: Week 4, 12, 24
SAS Result: Similar to Scenario 2 (fixed effect: treatment, visit, and treatment*visit)
Adjust means: -1.4852
Difference in adjust means: -1.4069. 95%CI (-1.6904,-1.1234)

R Result: Not Applicable. Because the template is having the dataset group by VISIT first, then fit ancova function for each sub-group dataset. In this case, the interaction (treatment*visit) cannot be added to the fixed effect.

Update description:
The interaction (treatment*visit) as a fixed effect can be added in s_ancova and h_ancova. But have a problem when added into summarize_ancova. 

# Findings:
1.	The results of ancova analysis using source data with two arms or multiple arms in 1 VISIT have no differences between R and SAS.
2.	If source data only contain one VISIT, model with fixed effects (treatment, visit and treatment*visit) has the same result as the model with fixed effect (treatment) in SAS;
3.	R cannot fit ancova model for source data with multiple VISITs (treatment, visit and treatment*visit). The results shown in multiple VISITs are actually using only treatment as the fixed effect. 

### sessionInfo()

```R
R version 4.0.4 (2021-02-15)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux 8.4 (Ootpa)

Matrix products: default
BLAS:   /usr/lib64/libblas.so.3.8.0
LAPACK: /opt/mango/R/4.0.4/lib/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] tern_0.7.6     rtables_0.4.1  magrittr_2.0.1

loaded via a namespace (and not attached):
 [1] pillar_1.6.0        compiler_4.0.4      tools_4.0.4         digest_0.6.27       lifecycle_1.0.0    
 [6] tibble_3.1.1        gtable_0.3.0        lattice_0.20-41     pkgconfig_2.0.3     rlang_0.4.10       
[11] Matrix_1.3-2        DBI_1.1.1           dplyr_1.0.5         generics_0.1.0      vctrs_0.3.7        
[16] grid_4.0.4          tidyselect_1.1.0    glue_1.4.2          R6_2.5.0            fansi_0.4.2        
[21] survival_3.2-10     dfoptim_2020.10-1   ggplot2_3.3.3       purrr_0.3.4         tidyr_1.1.3        
[26] scales_1.1.1        backports_1.2.1     ellipsis_0.3.1      htmltools_0.5.1.1   splines_4.0.4      
[31] assertthat_0.2.1    colorspace_2.0-0    numDeriv_2016.8-1.1 utf8_1.2.1          optimx_2020-4.2    
[36] munsell_0.5.0       broom_0.7.6         crayon_1.4.1
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

**Comment by Melkiades** [comment]
I was testing tern for other reasons and now the following is failing:
```r
  result <- h_ancova(
    .var = "Sepal.Length",
    .df_row = iris,
    variables = list(arm = "Species", covariates = c("Petal.Length * Petal.Width", "Sepal.Width"))
  ) %>%
    as.data.frame()

  expected <- structure(
    data.frame(
      Species = as.factor(c("setosa", "versicolor", "virginica")),
      emmean = c(6.154823, 5.717409, 5.406377),
      SE = c(0.33709131, 0.06680849, 0.14879463),
      df = c(143, 143, 143),
      lower.CL = c(5.488497, 5.585349, 5.112255),
      upper.CL = c(6.821149, 5.849469, 5.700498)
    ),
    estName = "emmean",
    clNames = c("lower.CL", "upper.CL"),
    pri.vars = "Species",
    adjust = "none",
    side = 0,
    delta = 0,
    type = "link",
    mesg = "Confidence level used: 0.95"
  )

  testthat::expect_equal(result, expected, tolerance = 0.0000001)
```
with the following output:
![Screenshot 2022-08-19 152346](https://user-images.githubusercontent.com/11279768/185628338-424c5746-7495-43ac-a494-edebb8555b3c.png)

**Comment by JiaLiu0001** [comment]
> ### What happened?
> ANCOVA model with treatment, visit, and treatment * visit interaction effect as fixed effects, and baseline as covariates. Different source data and tools might cause different results.
> 
> Methods in different tools: SAS: Proc mixed R: Ancova tern package ([AVOT01](https://docs.roche.com/doc/NEST%20Exploratory%20Releases/2022_01_28/tlg-catalog/))
> 
> Take treatment vs. CONTROL, Week 24 result as an example. The following are 3 scenarios to fit the model:
> 
> # 1. Two treatments: TRT & CONTROL. One visit: Week 24
> SAS Result: (fixed effect: treatment, visit, and treatment*visit) Adjust means: -1.4906 Difference in adjust means: -1.4086. 95%CI (-1.6070,-1.2103)
> 
> R Result: Adjust means: -1.49 Difference in adjust means: -1.41. 95%CI (-1.6,-1.22)
> 
> # 2. Multiple treatments: TRT2/3/4/6 & CONTROL. One visit: Week 24
> SAS Result: (fixed effect: treatment, visit, and treatment*visit) Adjust means: -1.4855 Difference in adjust means: -1.4071. 95%CI (-1.7373,-1.0768)
> 
> SAS model that has only treatment as the fixed effect Adjust means: -1.4855 Difference in adjust means: -1.4071. 95%CI (-1.7373,-1.0768)
> 
> R Result: Adjust means: -1.48 Difference in adjust means: -1.41. 95%CI (-1.71,-1.1)
> 
> # 3. Multiple treatments: TRT2/3/4/6 & CONTROL. Multiple visits: Week 4, 12, 24
> SAS Result: Similar to Scenario 2 (fixed effect: treatment, visit, and treatment*visit) Adjust means: -1.4852 Difference in adjust means: -1.4069. 95%CI (-1.6904,-1.1234)
> 
> R Result: Not Applicable. Because the template is having the dataset group by VISIT first, then fit ancova function for each sub-group dataset.
> 
> # Findings:
> 1. SAS and R have very similar ancova analysis results with the same source data;
> 2. The results of ancova analysis using source data with two arms or multiple arms have differences, especially in the 95% CI;.
> 3. If source data only contain one VISIT, model with fixed effects (treatment, visit and treatment*visit) has the same result as the model with fixed effect (treatment) in SAS;
> 4. R cannot fit ancova model for source data with multiple VISITs (treatment, visit and treatment*visit). The results shown in multiple visits are actually using only treatment as the fixed effect.
> 
> ### sessionInfo()
> ```r
> R version 4.0.4 (2021-02-15)
> Platform: x86_64-redhat-linux-gnu (64-bit)
> Running under: Red Hat Enterprise Linux 8.4 (Ootpa)
> 
> Matrix products: default
> BLAS:   /usr/lib64/libblas.so.3.8.0
> LAPACK: /opt/mango/R/4.0.4/lib/R/lib/libRlapack.so
> 
> locale:
>  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
>  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 
>  [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
> 
> attached base packages:
> [1] stats     graphics  grDevices utils     datasets  methods   base     
> 
> other attached packages:
> [1] tern_0.7.6     rtables_0.4.1  magrittr_2.0.1
> 
> loaded via a namespace (and not attached):
>  [1] pillar_1.6.0        compiler_4.0.4      tools_4.0.4         digest_0.6.27       lifecycle_1.0.0    
>  [6] tibble_3.1.1        gtable_0.3.0        lattice_0.20-41     pkgconfig_2.0.3     rlang_0.4.10       
> [11] Matrix_1.3-2        DBI_1.1.1           dplyr_1.0.5         generics_0.1.0      vctrs_0.3.7        
> [16] grid_4.0.4          tidyselect_1.1.0    glue_1.4.2          R6_2.5.0            fansi_0.4.2        
> [21] survival_3.2-10     dfoptim_2020.10-1   ggplot2_3.3.3       purrr_0.3.4         tidyr_1.1.3        
> [26] scales_1.1.1        backports_1.2.1     ellipsis_0.3.1      htmltools_0.5.1.1   splines_4.0.4      
> [31] assertthat_0.2.1    colorspace_2.0-0    numDeriv_2016.8-1.1 utf8_1.2.1          optimx_2020-4.2    
> [36] munsell_0.5.0       broom_0.7.6         crayon_1.4.1
> ```
> 
> ### Relevant log output
> _No response_
> 
> ### Code of Conduct
> * [x]  I agree to follow this project's Code of Conduct.
> 
> ### Contribution Guidelines
> * [x]  I agree to follow this project's Contribution Guidelines.
> 
> ### Security Policy
> * [x]  I agree to follow this project's Security Policy.

Update the description of the issue:
1. According to Ruchi, when it comes to only 2 treatments in comparison, ANCOVA is not the first choice. So scenario 1 above should be dropped out when considering this issue.
2. For scenario 3, 'h_ancova', 's_ancova', and 'a_ancova' can make it to include interaction fixed effects (treatment*visit). However, it's still a problem to add it into 'summarize_ancova'.
